src/Microsoft.DotNet.Compiler.Common/AssemblyInfoFileGenerator.cs

```diff
@@ -68,9 +68,15 @@ public static string GenerateFSharp(AssemblyInfoOptions metadata)
                 [typeof(AssemblyVersionAttribute)] = EscapeCharacters(metadata.AssemblyVersion?.ToString()),
                 [typeof(AssemblyInformationalVersionAttribute)] = EscapeCharacters(metadata.InformationalVersion),
                 [typeof(AssemblyCultureAttribute)] = EscapeCharacters(metadata.Culture),
-                [typeof(NeutralResourcesLanguageAttribute)] = EscapeCharacters(metadata.NeutralLanguage),
-                [typeof(TargetFrameworkAttribute)] = EscapeCharacters(metadata.TargetFramework)
+                [typeof(NeutralResourcesLanguageAttribute)] = EscapeCharacters(metadata.NeutralLanguage)
             };
+            
+            // only .net 4.0+ compatible
+            if (metadata.TargetFrameworkVersion == null || metadata.TargetFrameworkVersion >= new Version(4, 0))
```

Comment:
This isn't right. You need to parse the target framework as a NuGetFramework. The version isn't enough to tell. You need to make sure it's `IsDesktop()`  as well.

This also needs a test